File: src/linkerz_crawler_core/src/main/java/org/linkerz/crawler/core/parser/Parser.java
/*
 * Copyright (C) 2012 - 2013 LinkerZ
 */

package org.linkerz.crawler.core.parser;

import org.linkerz.crawler.core.downloader.result.DownloadResult;
import org.linkerz.crawler.core.parser.result.ParserResult;

import java.io.Serializable;

/**
 * The Class Parser.
 *
 * @author Nguyen Duc Dung
 * @since 7/2/12, 12:25 AM
 */
public interface Parser<DR extends DownloadResult, PR extends ParserResult> extends Serializable {
    PR parse(DR downloadResult);
}

File: src/linkerz_crawler_core/src/main/scala/org/linkerz/crawler/core/parser/Parser.scala
/*
 * Copyright (C) 2012 - 2013 LinkerZ (Searching and Sharing)
 */

package org.linkerz.crawler.core.parser

import grizzled.slf4j.Logging
import org.linkerz.crawler.core.job.CrawlJob


/**
 * The Class Parser.
 *
 * @author Nguyen Duc Dung
 * @since 7/29/12, 1:09 AM
 *
 */

trait Parser extends Logging {

  def parse(crawlJob: CrawlJob)

}